packages/firebase_database/lib/ui/firebase_list.dart

```diff
@@ -13,102 +14,40 @@ typedef void ChildCallback(int index, DataSnapshot snapshot);
 typedef void ChildMovedCallback(int fromIndex, int toIndex, DataSnapshot snapshot);
 typedef void ValueCallback(DataSnapshot snapshot);
 
-/// Sorts the results of `query` on the client side using `DataSnapshot.key`.
-class FirebaseList extends ListBase<DataSnapshot> with StreamSubscriberMixin<Event> {
+/// Base class for a list that tracks the results of a query.
+///
+/// You must call dispose() when the list is no longer used.
+///
+/// See also:
+/// * ClientSortedList
+/// * ServerSortedList
+abstract class FirebaseList extends ListBase<DataSnapshot> with StreamSubscriberMixin<Event> {
```

Comment:
dispose() won't get called in the StreamSubscriberMixin class, so it will leak the subscriptions.